android/app/src/main/java/com/microsoft/codepush/react/CodePushNativeModule.java

```diff
@@ -124,24 +119,20 @@ private void setJSBundle(ReactInstanceManager instanceManager, String latestJSBu
             if (numParameters == 1) {
                 // RN >= v0.34
                 latestJSBundleLoader = createFileLoaderMethod.invoke(jsBundleLoaderClass, latestJSBundleFile);
-            } else if (numParameters == 2) {
-                // RN >= v0.31 && RN < v0.34
-                latestJSBundleLoader = createFileLoaderMethod.invoke(jsBundleLoaderClass, getReactApplicationContext(), latestJSBundleFile);
             } else {
                 throw new NoSuchMethodException("Could not find a recognized 'createFileLoader' method");
             }
 
             bundleLoaderField.setAccessible(true);
             bundleLoaderField.set(instanceManager, latestJSBundleLoader);
         } catch (Exception e) {
-            // RN < v0.31
-            Field jsBundleField = instanceManager.getClass().getDeclaredField("mJSBundleFile");
-            jsBundleField.setAccessible(true);
-            jsBundleField.set(instanceManager, latestJSBundleFile);
+            CodePushUtils.log("Unable to set JSBundle - CodePush may not support this version of React Native");
+            throw new IllegalAccessException("Could not setJSBundle");
         }
     }
 
     private void loadBundle() {
+        clearLifecycleEventListener();
```

Comment:
Any issue with clearing this before we attempt to load the bundle? We don't need the callback after this point.

We must call this before `loadBundleLegacy()` or it will go into an infinite restart loop.